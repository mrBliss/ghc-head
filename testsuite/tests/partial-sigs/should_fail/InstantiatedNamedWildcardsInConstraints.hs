{-# LANGUAGE NamedWildcards #-}
module InstantiatedNamedWildcardsInConstraints where

foo :: (Enum _a, _) => _a -> (String, b)
foo x = (show (succ x), x)

