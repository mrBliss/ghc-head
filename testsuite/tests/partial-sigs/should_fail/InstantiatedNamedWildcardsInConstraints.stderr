
InstantiatedNamedWildcardsInConstraints.hs:4:14:
    Found hole ‘_a’ with type: b
    Where: ‘b’ is a rigid type variable bound by
               the inferred type of foo :: (Enum b, Show b) => b -> (String, b)
               at InstantiatedNamedWildcardsInConstraints.hs:4:8
    To use the inferred type, enable PartialTypeSignatures
    In the type signature for ‘foo’:
      foo :: (Enum _a, _) => _a -> (String, b)

InstantiatedNamedWildcardsInConstraints.hs:4:18:
    Found hole ‘_’ with inferred constraints: (Show b)
    Where: ‘b’ is a rigid type variable bound by
               the inferred type of foo :: (Enum b, Show b) => b -> (String, b)
               at InstantiatedNamedWildcardsInConstraints.hs:4:8
    To use the inferred type, enable PartialTypeSignatures
    In the type signature for ‘foo’:
      foo :: (Enum _a, _) => _a -> (String, b)

InstantiatedNamedWildcardsInConstraints.hs:4:24:
    Found hole ‘_a’ with type: b
    Where: ‘b’ is a rigid type variable bound by
               the inferred type of foo :: (Enum b, Show b) => b -> (String, b)
               at InstantiatedNamedWildcardsInConstraints.hs:4:8
    To use the inferred type, enable PartialTypeSignatures
    In the type signature for ‘foo’:
      foo :: (Enum _a, _) => _a -> (String, b)
